### DISCUSSION TITLE
Fix the Windows 10 Start Menu and add an option to modernize the Windows 10 Start Menu further by using XamlDiagnostics.

### DISCUSSION THREAD

**Discussion by lixkote** [valinet/ExplorerPatcher#1111] [Ideas]
So the issue is, on the latest 22621, the UWP apps are no longer displaying in the start menu, could you fix that? 
You can add an option to achieve look like that by using XamlDiagnostics as i did in my prototype app. 
![image-2](https://user-images.githubusercontent.com/95425619/226120284-3f35656d-f2c7-4db6-80fe-7001249a3757.png)

**Comment by valinet** [comment]
Let's not open duplicate threads, why does almost everyone feel the need to do that? It's easier when information on similar topics is relatively centralized in a single thread... 

https://github.com/valinet/ExplorerPatcher/issues/1108

**Reply by valinet** [reply]
Btw, what is XamlDiagnostics? Can you be a bit more verbose? Throwing 2 lines and a screenshot, very informative...

**Reply by lixkote** [reply]
I'm sorry I didn't explain what it was. Afaik generally speaking, XamlDiagnostics is a WinRT Api that allows you to debug Xaml content. You might be interested in the API documented here, it's basically the official capture API for XAML (WUX):
https://github.com/MicrosoftDocs/sdk-api/pull/1248
this allows you to easily change the XAML content of your application without having to use symbols and/or non-exported functions, here is an example of this API in action made by someone named Kimbra:

https://user-images.githubusercontent.com/95425619/226133852-9cdae6e8-6ac4-485f-b3d1-fe8f7842a3ee.mp4

**Reply by valinet** [reply]
Yeah, looks interesting, indeed. I personally learned how to hide the "Recommended" section by parsing the visual tree and adjusting properties there - it doesn't require symbols either, but indeed, from what I can see in your video, this API seems to be very powerful and allows one to achieve plenty things. Where is this guy's work, is it open source, can we learn from it? Videos look nice, but actual knowledge is even better.

**Reply by lixkote** [reply]
Do you have a discord acc by any chance?
I know his account on there, but he does not want to accept my dm......
